pkg/mutation/annotations.go

```diff
@@ -0,0 +1,36 @@
+package mutation
+
+import (
+	"strings"
+
+	"github.com/google/uuid"
+	"github.com/open-policy-agent/gatekeeper/pkg/mutation/types"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+const (
+	annotationMutations  = "gatekeeper.sh/mutations"
+	annotationMutationID = "gatekeeper.sh/mutation-id"
+)
+
+func mutationAnnotations(obj *unstructured.Unstructured, allAppliedMutations [][]types.Mutator, mutationUUID uuid.UUID) {
+	mutatorStringSet := make(map[string]struct{})
+	for _, mutationsForIteration := range allAppliedMutations {
+		for _, mutator := range mutationsForIteration {
+			mutatorStringSet[mutator.String()] = struct{}{}
```

Comment:
why do we need mutatorStringSet here? can we append to mutatorStrings directly?

Reply:
This preserves the original logic.

Since mutations are applied each time, you can end up with duplicates in the case of oscillations which terminate.

Reply:
(This is just a copy+paste of the original logic)